sonic_installer/main.py

```diff
@@ -334,6 +337,32 @@ def install(url, force):
         # copy directories and preserve original file structure, attributes and associated metadata
         run_command("cp -ar /etc/sonic /host/old_config")
 
+    # Check Latest BIOS availability
+    try:
+        import sonic_platform.chassis
+        component_name = "BIOS"
+        chassis = sonic_platform.chassis.Chassis()
+
+        next_image = get_next_image()
+        squashfs_path = HOST_PATH + '/' + \
+                next_image.replace(IMAGE_PREFIX, IMAGE_DIR_PREFIX) + \
+                '/' + 'fs.squashfs'
+        if squashfs_path:
+            tmpdir = tempfile.mkdtemp()
+            run_command(
+                'mount ' + squashfs_path + ' ' + tmpdir + \
+                ' -t squashfs -o loop')
+            status = chassis.install_component_firmware(component_name, tmpdir)
```

Comment:
I do not see the bios firmware name in the tmpdir. should that be a filename?

Reply:
- Each vendor, do have a different BIOS filename. So, if we attempt to pass a filename instead of tmpdir we end-up with multiple combinations.  To avoid this, we pass only mounted squashfs location as tmpdir. 
- Now, individual vendors do construct their BIOS image path according to their naming standards form the mounted filesystem. You can see the implementation below:
- https://github.com/Azure/sonic-buildimage/blob/0d8c1b2463a677a477e76ef395ae64569e43508f/platform/broadcom/sonic-platform-modules-dell/s6100/sonic_platform/chassis.py#L161

Reply:
- Each vendor, do have a different BIOS filename. So, if we attempt to pass a filename instead of tmpdir we end-up with multiple combinations.  To avoid this, we pass only mounted squashfs location as tmpdir. 
- Now, individual vendors do construct their BIOS image path according to their naming standards form the mounted filesystem. You can see the implementation below:
- https://github.com/Azure/sonic-buildimage/blob/0d8c1b2463a677a477e76ef395ae64569e43508f/platform/broadcom/sonic-platform-modules-dell/s6100/sonic_platform/chassis.py#L161

Reply:
it should be at least under /usr/sonic/dev/{platform} folder.

Reply:
- It is already addresed. DellEMC BIOS file do reside in following directory:
         /usr/share/sonic/device/x86_64-dell_s6100_c2538-r0/bin/ [BIOS filepath](https://github.com/Azure/sonic-buildimage/blob/0d8c1b2463a677a477e76ef395ae64569e43508f/platform/broadcom/sonic-platform-modules-dell/s6100/sonic_platform/chassis.py#L145)
- BIOS filename is fetched from chassis.py. [BIOS filename](https://github.com/Azure/sonic-buildimage/blob/0d8c1b2463a677a477e76ef395ae64569e43508f/platform/broadcom/sonic-platform-modules-dell/s6100/sonic_platform/chassis.py#L140)
- Do you need any change here?